manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
```

Comment:
Can it be customized? I think most people (I do anyway) prefer their particular git habits and don't want to be forced into a way of doing things.